### PR TITLE
Add docs for `devcontainer templates publish`

### PR DESCRIPTION
**Pull request by joshspicer**
ref: https://github.com/microsoft/vscode-dev-containers/pull/1238

### CODE REVIEW

File: docs/templates/publish.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Publishing Dev Container Templates
+
+> NOTE: You may want to first check out our [templates-starter](https://github.com/devcontainers/template-starter), which includes an example [actions workflow](https://github.com/devcontainers/action) for publishing directly out of your GitHub repo!
+
+## Summary
+
+The CLI can be used publish [Dev Container Template](https://containers.dev/implementors/templates/) artifacts to an OCI registry (that supports the [artifacts specification](https://oras.land/implementors/)).
```
Comment (samruddhikhandale) [inline docs/templates/publish.md:7]:
```suggestion
The CLI can be used to publish [Dev Container Template](https://containers.dev/implementors/templates/) artifacts to an OCI registry (that supports the [artifacts specification](https://oras.land/implementors/)).
```

**Review by samruddhikhandale — DISMISSED** [review]
Nice, thank you! ⚡

File: docs/templates/publish.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+> NOTE: OS-specific docker credential helpers (Docker Desktop credential helper) are not currently recognized by the CLI.  
+
+- Adding a $HOME/.docker/config.json with your credentials following [this commonly defined format](https://www.systutorials.com/docs/linux/man/5-docker-config-json/).
+   - Your `docker login` command may write this file for you depending on your operating system.
+- Using our custom env variable DEVCONTAINERS_OCI_AUTH
+    - eg: `DEVCONTAINERS_OCI_AUTH=service1|user1|token1,service2|user2|token2`
+    
+For publishing to `ghcr.io`
+- Using the `devcontainers/action` GitHub action to handle the `GITHUB_TOKEN` credential for you.
+- Providing a GITHUB_TOKEN with permission to `write:packages`,
```
Comment (bamurtaugh) [inline docs/templates/publish.md:65]:
```suggestion
- Providing a GITHUB_TOKEN with permission to `write:packages`.
```

**Review by bamurtaugh — APPROVED** [review]
Thanks for documenting this, @joshspicer!

I opened a PR for containers.dev: https://github.com/devcontainers/devcontainers.github.io/pull/152.

And I'll also open a PR to point to this info from the VS Code docs.

**Pull request merged**